### PR TITLE
jv-spring-rest

### PR DESCRIPTION
**Pull request by morozovyrv**

### CODE REVIEW

File: src/main/java/mate/academy/spring/controller/CinemaHallController.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    private final CinemaHallDtoMapper cinemaHallDtoMapper;
+
+    public CinemaHallController(CinemaHallService cinemaHallService,
+                                CinemaHallDtoMapper cinemaHallDtoMapper) {
+        this.cinemaHallService = cinemaHallService;
+        this.cinemaHallDtoMapper = cinemaHallDtoMapper;
+    }
+
+    @GetMapping("/cinema-halls")
+    @ResponseStatus(HttpStatus.OK)
```
Comment (aaananas) [inline src/main/java/mate/academy/spring/controller/CinemaHallController.java:28]:
```suggestion
```
That was not in the task

File: src/main/java/mate/academy/spring/controller/CinemaHallController.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    public List<CinemaHallResponseDto> getAll() {
+        return cinemaHallService.getAll()
+                .stream()
+                .map(cinemaHallDtoMapper::toDto)
+                .collect(Collectors.toList());
+    }
+
+    @PostMapping("/cinema-halls")
+    @ResponseStatus(HttpStatus.CREATED)
```
Comment (aaananas) [inline src/main/java/mate/academy/spring/controller/CinemaHallController.java:37]:
```suggestion
```

File: src/main/java/mate/academy/spring/controller/MovieController.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    private final MovieDtoMapper movieDtoMapper;
+
+    public MovieController(MovieService movieService,
+                           MovieDtoMapper movieDtoMapper) {
+        this.movieService = movieService;
+        this.movieDtoMapper = movieDtoMapper;
+    }
+
+    @PostMapping("/movies")
+    @ResponseStatus(HttpStatus.CREATED)
```
Comment (aaananas) [inline src/main/java/mate/academy/spring/controller/MovieController.java:28]:
```suggestion
```

File: src/main/java/mate/academy/spring/controller/MovieController.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    public MovieResponseDto create(@RequestBody MovieRequestDto requestDto) {
+        return movieDtoMapper.toDto(
+                movieService.add(
+                        movieDtoMapper.toModel(requestDto)
+                )
+        );
+    }
+
+    @GetMapping("/movies")
+    @ResponseStatus(HttpStatus.OK)
```
Comment (aaananas) [inline src/main/java/mate/academy/spring/controller/MovieController.java:38]:
```suggestion
```

File: src/main/java/mate/academy/spring/controller/MovieSessionController.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    private final MovieSessionDtoMapper movieSessionDtoMapper;
+
+    public MovieSessionController(MovieSessionService movieSessionService,
+                                  MovieSessionDtoMapper movieSessionDtoMapper) {
+        this.movieSessionService = movieSessionService;
+        this.movieSessionDtoMapper = movieSessionDtoMapper;
+    }
+
+    @PostMapping("/movie-sessions")
+    @ResponseStatus(HttpStatus.CREATED)
```
Comment (aaananas) [inline src/main/java/mate/academy/spring/controller/MovieSessionController.java:35]:
```suggestion
```

File: src/main/java/mate/academy/spring/controller/MovieSessionController.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+                                              MovieSessionRequestDto requestDto) {
+        return movieSessionDtoMapper.toDto(
+                movieSessionService.add(
+                        movieSessionDtoMapper.toModel(requestDto)
+                )
+        );
+    }
+
+    @GetMapping("/movie-sessions/available")
+    @ResponseStatus(HttpStatus.OK)
```
Comment (aaananas) [inline src/main/java/mate/academy/spring/controller/MovieSessionController.java:46]:
```suggestion
```

File: src/main/java/mate/academy/spring/controller/MovieSessionController.java [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+    public MovieSessionResponseDto update(@PathVariable Long id,
+                                          @RequestBody MovieSessionRequestDto requestDto) {
+        MovieSession movieSession = movieSessionDtoMapper.toModel(requestDto);
+        movieSession.setId(id);
+        return movieSessionDtoMapper.toDto(movieSessionService.update(movieSession));
+    }
+
+    @DeleteMapping("/movie-sessions/{id}")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
```
Comment (aaananas) [inline src/main/java/mate/academy/spring/controller/MovieSessionController.java:67]:
```suggestion
```

File: src/main/java/mate/academy/spring/controller/MovieSessionController.java [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+                                                                 (pattern = "dd.MM.yyyy")
+                                                         LocalDate date) {
+        return movieSessionService.findAvailableSessions(movieId, date)
+                .stream()
+                .map(movieSessionDtoMapper::toDto)
+                .collect(Collectors.toList());
+    }
+
+    @PutMapping("/movie-sessions/{id}")
+    @ResponseStatus(HttpStatus.OK)
```
Comment (aaananas) [inline src/main/java/mate/academy/spring/controller/MovieSessionController.java:58]:
```suggestion
```

File: pom.xml [hunk 9]
```diff
@@ -29,6 +30,48 @@
...
+        <dependency>
+            <groupId>com.fasterxml.jackson.core</groupId>
+            <artifactId>jackson-databind</artifactId>
+            <version>2.13.3</version>
+        </dependency>
+        <dependency>
+            <groupId>com.fasterxml.jackson.core</groupId>
+            <artifactId>jackson-databind</artifactId>
+            <version>2.13.3</version>
+        </dependency>
```
Comment (aaananas) [inline pom.xml:74]: duplicate
```suggestion
```

File: src/main/java/mate/academy/spring/controller/CinemaHallController.java [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+import mate.academy.spring.service.CinemaHallService;
+import mate.academy.spring.service.mapper.CinemaHallDtoMapper;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
```
Comment (aaananas) [inline src/main/java/mate/academy/spring/controller/CinemaHallController.java:16]:
```suggestion
@RestController("/cinema-halls")
```

File: src/main/java/mate/academy/spring/controller/CinemaHallController.java [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+    private final CinemaHallService cinemaHallService;
+    private final CinemaHallDtoMapper cinemaHallDtoMapper;
+
+    public CinemaHallController(CinemaHallService cinemaHallService,
+                                CinemaHallDtoMapper cinemaHallDtoMapper) {
+        this.cinemaHallService = cinemaHallService;
+        this.cinemaHallDtoMapper = cinemaHallDtoMapper;
+    }
+
+    @GetMapping("/cinema-halls")
```
Comment (aaananas) [inline src/main/java/mate/academy/spring/controller/CinemaHallController.java:27]:
```suggestion
    @GetMapping
```

File: src/main/java/mate/academy/spring/controller/CinemaHallController.java [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+    @ResponseStatus(HttpStatus.OK)
+    public List<CinemaHallResponseDto> getAll() {
+        return cinemaHallService.getAll()
+                .stream()
+                .map(cinemaHallDtoMapper::toDto)
+                .collect(Collectors.toList());
+    }
+
+    @PostMapping("/cinema-halls")
```
Comment (aaananas) [inline src/main/java/mate/academy/spring/controller/CinemaHallController.java:36]:
```suggestion
    @PostMapping
```

File: src/main/java/mate/academy/spring/controller/MovieController.java [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+import mate.academy.spring.service.MovieService;
+import mate.academy.spring.service.mapper.MovieDtoMapper;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
```
Comment (aaananas) [inline src/main/java/mate/academy/spring/controller/MovieController.java:16]:
```suggestion
@RestController("/movies")
```

File: src/main/java/mate/academy/spring/controller/MovieController.java [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+    private final MovieService movieService;
+    private final MovieDtoMapper movieDtoMapper;
+
+    public MovieController(MovieService movieService,
+                           MovieDtoMapper movieDtoMapper) {
+        this.movieService = movieService;
+        this.movieDtoMapper = movieDtoMapper;
+    }
+
+    @PostMapping("/movies")
```
Comment (aaananas) [inline src/main/java/mate/academy/spring/controller/MovieController.java:27]:
```suggestion
    @PostMapping
```

File: src/main/java/mate/academy/spring/controller/MovieController.java [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+    @ResponseStatus(HttpStatus.CREATED)
+    public MovieResponseDto create(@RequestBody MovieRequestDto requestDto) {
+        return movieDtoMapper.toDto(
+                movieService.add(
+                        movieDtoMapper.toModel(requestDto)
+                )
+        );
+    }
+
+    @GetMapping("/movies")
```
Comment (aaananas) [inline src/main/java/mate/academy/spring/controller/MovieController.java:37]:
```suggestion
    @GetMapping
```

File: src/main/java/mate/academy/spring/controller/MovieSessionController.java [hunk 16]
```diff
@@ -0,0 +1,71 @@
...
+    private final MovieSessionService movieSessionService;
+    private final MovieSessionDtoMapper movieSessionDtoMapper;
+
+    public MovieSessionController(MovieSessionService movieSessionService,
+                                  MovieSessionDtoMapper movieSessionDtoMapper) {
+        this.movieSessionService = movieSessionService;
+        this.movieSessionDtoMapper = movieSessionDtoMapper;
+    }
+
+    @PostMapping("/movie-sessions")
```
Comment (aaananas) [inline src/main/java/mate/academy/spring/controller/MovieSessionController.java:34]:
```suggestion
    @PostMapping
```

File: src/main/java/mate/academy/spring/controller/MovieSessionController.java [hunk 17]
```diff
@@ -0,0 +1,71 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
```
Comment (aaananas) [inline src/main/java/mate/academy/spring/controller/MovieSessionController.java:23]:
```suggestion
@RestController("/movie-sessions")
```

File: src/main/java/mate/academy/spring/controller/CinemaHallController.java [hunk 18]
```diff
@@ -0,0 +1,45 @@
...
+    @ResponseStatus(HttpStatus.CREATED)
+    public CinemaHallResponseDto create(@RequestBody CinemaHallRequestDto requestDto) {
+        return cinemaHallDtoMapper.toDto(
+                cinemaHallService.add(
+                        cinemaHallDtoMapper.toModel(requestDto)
+                )
+        );
```
Comment (aaananas) [inline src/main/java/mate/academy/spring/controller/CinemaHallController.java:43]:
```suggestion
        return cinemaHallDtoMapper.toDto(
                        cinemaHallService.add(cinemaHallDtoMapper.toModel(requestDto)));
```

File: src/main/java/mate/academy/spring/controller/MovieController.java [hunk 19]
```diff
@@ -0,0 +1,45 @@
...
+
+    @PostMapping("/movies")
+    @ResponseStatus(HttpStatus.CREATED)
+    public MovieResponseDto create(@RequestBody MovieRequestDto requestDto) {
+        return movieDtoMapper.toDto(
+                movieService.add(
+                        movieDtoMapper.toModel(requestDto)
+                )
+        );
+    }
```
Comment (aaananas) [inline src/main/java/mate/academy/spring/controller/MovieController.java:35]:
```suggestion
        return movieDtoMapper.toDto(
                movieService.add(movieDtoMapper.toModel(requestDto)));
    }
```

File: src/main/java/mate/academy/spring/controller/MovieSessionController.java [hunk 20]
```diff
@@ -0,0 +1,71 @@
...
+
+    @PostMapping("/movie-sessions")
+    @ResponseStatus(HttpStatus.CREATED)
+    public MovieSessionResponseDto create(@RequestBody
+                                              MovieSessionRequestDto requestDto) {
+        return movieSessionDtoMapper.toDto(
+                movieSessionService.add(
+                        movieSessionDtoMapper.toModel(requestDto)
+                )
+        );
```
Comment (aaananas) [inline src/main/java/mate/academy/spring/controller/MovieSessionController.java:42]:
```suggestion
        return movieSessionDtoMapper.toDto(
                movieSessionService.add(movieSessionDtoMapper.toModel(requestDto)));
```

File: src/main/java/mate/academy/spring/dao/impl/MovieSessionDaoImpl.java [hunk 21]
```diff
@@ -57,4 +59,52 @@ public Optional<MovieSession> get(Long id) {
...
+    public void delete(Long id) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            Query<MovieSession> query = session.createQuery("DELETE FROM "
+                    + "MovieSession ms WHERE ms.id = :id");
+            query.setParameter("id", id);
+            query.executeUpdate();
```
Comment (aaananas) [inline src/main/java/mate/academy/spring/dao/impl/MovieSessionDaoImpl.java:96]:
```suggestion
            session = sessionFactory.openSession();
            transaction = session.beginTransaction();
            session.delete(movieSession);
```

File: src/main/java/mate/academy/spring/dao/impl/MovieSessionDaoImpl.java [hunk 22]
```diff
@@ -57,4 +59,52 @@ public Optional<MovieSession> get(Long id) {
...
+                    + movieSession, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public void delete(Long id) {
```
Comment (aaananas) [inline src/main/java/mate/academy/spring/dao/impl/MovieSessionDaoImpl.java:87]:
```suggestion
    public void delete(MovieSession movieSession) {
```

**Review by aaananas — CHANGES_REQUESTED** [review]

**Review by aaananas — CHANGES_REQUESTED** [review]
fix previous requirements

**Review by zachar-chorny — CHANGES_REQUESTED** [review]
It's okay, but not all edits are fixed. Please carefully review what Anastasia wrote to you again

**Pull request closed without merging**